src/utils/useChromaticDialog.ts

```diff
@@ -0,0 +1,53 @@
+import { useCallback, useEffect, useRef } from "react";
+
+export type DialogEventName = "login";
+export type DialogPayload<T extends DialogEventName> = Record<string, never>;
+export type DialogHandlers = {
+  [Event in DialogEventName]?: (payload: DialogPayload<Event>) => void;
+};
+
+export const useChromaticDialog = (handlers: DialogHandlers = {}) => {
+  const dialog = useRef<Window>();
+  const dialogOrigin = useRef<string>();
+
+  // Close the dialog window when the screen gets unmounted.
+  useEffect(() => () => dialog.current?.close(), []);
+
+  useEffect(() => {
+    const handleMessage = ({ origin, data }: MessageEvent) => {
+      if (origin === dialogOrigin.current) {
+        const { message, ...rest } = data;
+        if (message in handlers) {
+          handlers[message as DialogEventName](rest);
+        }
+      }
+    };
+
+    window.addEventListener("message", handleMessage, false);
+
+    return () => {
+      window.removeEventListener("message", handleMessage);
+    };
+  }, [handlers, dialogOrigin]);
+
+  return useCallback(
+    (url: string) => {
+      const width = 800;
+      const height = 800;
+      const usePopup = window.innerWidth > width && window.innerHeight > height;
+
+      if (usePopup) {
+        const left = (window.innerWidth - width) / 2 + window.screenLeft;
+        const top = (window.innerHeight - height) / 2 + window.screenTop;
+        const options = `scrollbars=yes,width=${width},height=${height},top=${top},left=${left}`;
+        dialog.current = window.open(url, "oauth-dialog", options);
```

Comment:
```suggestion
        dialog.current = window.open(url, "chromatic-dialog", options);
```
We could make this an option to `useChromaticDialog`.

Reply:
I'll wait until we need it to add that.